daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -468,6 +461,66 @@ class DFDLTestSuite private[tdml] (
     }
   }
 
+  /**
+   * The CompileResult cache is stored in the DFDLTestSuite instance. This
+   * means any tests run from the same instance can share the same compiled
+   * processor. Once the DFDLTestSuite goes out of scope and is garbage
+   * collected, so too will be the compiled processors. Running the tests again
+   * will require creating a new DFDLTestSuite instance, and thus will
+   * recompile the schema.
+   */
+  private val tdmlCompileResultCache = new mutable.HashMap[TDMLCompileResultCacheKey, TDML.CompileResult]()
```

Comment:
HashMap has a companion object with an apply method so I think you can remove the `new`.

def apply[K, V](elems: (K, V)*): HashMap[K, V]
A collection of type Map that contains given key/value bindings.